internal/extproc/chatcompletion_processor.go

```diff
@@ -70,6 +78,22 @@ type chatCompletionProcessorRouterFilter struct {
 	upstreamFilterCount int
 }
 
+// ProcessResponseHeaders implements [Processor.ProcessResponseHeaders].
+func (c *chatCompletionProcessorRouterFilter) ProcessResponseHeaders(ctx context.Context, headerMap *corev3.HeaderMap) (*extprocv3.ProcessingResponse, error) {
+	if c.upstreamFilter != nil { // See the comment on the "upstreamFilter" field.
+		return c.upstreamFilter.ProcessResponseHeaders(ctx, headerMap)
+	}
+	return c.passThroughProcessor.ProcessResponseHeaders(ctx, headerMap)
```

Comment:
Why do we need to call passThroughProcessor ?

Reply:
https://github.com/envoyproxy/ai-gateway/pull/644/commits/67b9e1de6f6249972f9121b72df2fb57cc9ae944 good q and added comment